src/Features/Core/Portable/Diagnostics/EngineV2/CompilerDiagnosticExecutor.cs

```diff
@@ -61,7 +61,7 @@ internal static class CompilerDiagnosticExecutor
                 builder.Add(analyzer, result.ToResult());
             }
 
-            return builder.ToImmutable();
+            return new CompilerAnalysisResult(builder.ToImmutable(), analysisResult.AnalyzerTelemetryInfo);
```

Comment:
return telemetry info as well as analysis info. new API change made existing API in compilation not to work.